src/mscorlib/shared/System/Span.NonGeneric.cs

```diff
@@ -249,6 +423,128 @@ internal static unsafe bool OrdinalHelper(ReadOnlySpan<char> span, ReadOnlySpan<
             }
         }
 
+        private static unsafe int CompareOrdinalHelper(ReadOnlySpan<char> span, ReadOnlySpan<char> value)
+        {
+            Debug.Assert(value.Length != 0);
+            Debug.Assert(span.Length != 0);
+            int length = Math.Min(span.Length, value.Length);
+
+            fixed (char* ap = &MemoryMarshal.GetReference(span))
+            fixed (char* bp = &MemoryMarshal.GetReference(value))
+            {
+                char* a = ap;
+                char* b = bp;
+
+                if (*a != *b)
+                    goto DiffOffsetStart;
+
+                if (length >= 2)
+                {
+                    if (*(a + 1) != *(b + 1))
+                        goto DiffOffset1;
+
+                    // Since we know that the first two chars are the same,
+                    // we can increment by 2 here and skip 4 bytes.
+                    // This leaves us 8-byte aligned, which results
```

Comment:
>This leaves us 8-byte [](start = 23, length = 21)

@jkotas does 4-bytes alignments considered as 8-bytes aligned too?

Reply:
This comment and optimization was true for `String` implementation. It does not hold for `Span` implementation.